zerver/tests/test_users.py

```diff
@@ -2246,6 +2247,87 @@ def test_do_delete_user(self) -> None:
                 ).exists()
             )
 
+    def test_do_delete_user_preserving_messages(self) -> None:
```

Comment:
This test is uncomfortably almost-identical to `test_do_delete_user` right above it - but not sure how to reasonably refactor to not duplicate the code. At least the initial setup of messages could be extracted into a helper method, but then it'd have to return several integer lists like `huddle_message_ids_from_cordelia`, `huddle_message_ids_from_hamlet`. `personal_message_ids_to_hamlet`, which might be even uglier

Reply:
It's probably OK, I think.